File: movieShop/movieShopServer/src/main/java/lab/movieShopServer/models/Movie.java
package lab.movieShopServer.models;
// https://www.timeout.com/film/best-movies-of-all-time

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

@Entity
public class Movie {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @NotEmpty
    private String title;

    @NotEmpty
    @Column(length = 2000)
    private String description;

    @NotNull
    private Long releaseYear;

    @NotEmpty
    private String genre;

    @NotEmpty
    private String director;

    @NotNull
    @Min(value = 0)
    private Double price;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Long getReleaseYear() {
        return releaseYear;
    }

    public void setReleaseYear(Long releaseYear) {
        this.releaseYear = releaseYear;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public String getDirector() {
        return director;
    }

    public void setDirector(String director) {
        this.director = director;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }
}

File: movieShop/movieShopFront/src/app/movie.ts
export class Movie {
    id: number;
    title: string;
    description: string;
    releaseYear: number;
    genre: string;
    director: string;
    price: number;

    constructor(
        id: number,
        title: string,
        description: string,
        releaseYear: number,
        genre: string,
        director: string,
        price: number,
    ){
        this.title = title;
        this.description = description;
        this.releaseYear = releaseYear;
        this.genre = genre;
        this.director = director;
        this.price =price;
        this.id = id;
    }
}